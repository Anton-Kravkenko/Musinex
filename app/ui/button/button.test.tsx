import Button from '@/ui/button/button'
import {
	cleanup,
	fireEvent,
	render,
	screen,
	waitFor
} from '@testing-library/react-native'

afterEach(() => cleanup())
const mockFn = jest.fn()
describe('Button', () => {
	it('Render', function () {
		const buttonRender = render(
			<Button
				text={'by'}
				size={'large'}
				style={{
					backgroundColor: 'red'
				}}
				onPress={mockFn}
				className='mb-4'
				icon={'search'}
			/>
		)
		expect(buttonRender).toMatchSnapshot()
	})

	it('OnPress', () => {
		render(
			<Button text={'by'} size={'large'} onPress={mockFn} icon={'search'} />
		)
		expect(screen.getByTestId('button')).toBeDefined()
		fireEvent.press(screen.getByTestId('button'))
		expect(mockFn).toHaveBeenCalledTimes(1)
	})
	it('Check Children', function () {
		render(
			<Button text={'by'} size={'large'} onPress={mockFn} icon={'search'} />
		)
		expect(screen.getByTestId('button')).toBeDefined()
		expect(screen.getByTestId('button').children[0]).toBeDefined()
	})
	it('Check Text', async function () {
		const { getByTestId, queryByTestId } = render(
			<Button text={'by'} size={'large'} onPress={mockFn} icon={'search'} />
		)
		await waitFor(() => expect(queryByTestId('skeleton')).toBeNull())
		expect(getByTestId('title')).toHaveTextContent('by')
	})
})
