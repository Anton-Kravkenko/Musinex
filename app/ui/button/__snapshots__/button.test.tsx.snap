// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  className="mb-4"
  collapsable={false}
  focusable={true}
  icon="search"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#888888",
        "borderRadius": 10,
        "padding": 12,
        "width": "100%",
      },
      {
        "backgroundColor": "red",
      },
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      },
    ]
  }
  testID="button"
  text="by"
>
  <View
    className="flex-row items-center"
    style={
      {
        "gap": 10,
        "justifyContent": "space-between",
        "marginVertical": 5,
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#888888",
            "borderRadius": 6,
            "height": 20,
            "width": 20,
          },
          {
            "letterSpacing": 1,
            "textTransform": "none",
          },
        ]
      }
      testID="skeleton"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#000",
            "borderRadius": 16,
            "borderWidth": 0,
            "justifyContent": "center",
            "padding": 8,
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
      testID="icon-wrapper"
    >
      <Text />
    </View>
  </View>
</View>
`;
